packages/flame/lib/src/components/input/advanced_button_component.dart

```diff
@@ -0,0 +1,292 @@
+import 'package:flame/components.dart';
+import 'package:flame/events.dart';
+import 'package:flame/layout.dart';
+import 'package:flutter/foundation.dart';
+
+/// The [AdvancedButtonComponent] has different skins for
+/// different button states.
+/// The defaultSkin must be added to the constructor or
+/// if you are inheriting - defined in the onLod method
+///
+///The label is a [PositionComponent] and is added
+///to the foreground of the button. The label is aligned to the center.
+///
+/// Note: You have to set the [defaultSkin],[downSkin],[hoverSkin]
+/// [disabledSkin], [defaultLabel] in [onLoad] if you are
+/// not passing it in through the constructor
+
+class AdvancedButtonComponent extends PositionComponent
+    with HoverCallbacks, TapCallbacks {
+  AdvancedButtonComponent({
+    this.onPressed,
+    this.onChangeState,
+    PositionComponent? defaultSkin,
+    PositionComponent? downSkin,
+    PositionComponent? hoverSkin,
+    PositionComponent? disabledSkin,
+    PositionComponent? defaultLabel,
+    PositionComponent? disabledLabel,
+    super.size,
+    super.position,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+  }) {
+    this.defaultSkin = defaultSkin;
+    this.downSkin = downSkin;
+    this.hoverSkin = hoverSkin;
+    this.disabledSkin = disabledSkin;
+    this.defaultLabel = defaultLabel;
+    this.disabledLabel = disabledLabel;
+    size.addListener(_updateSizes);
+  }
+
+  /// Callback for what should happen when the button is pressed.
+  void Function()? onPressed;
+
+  /// Callback when button state changes
+  void Function(ButtonState state)? onChangeState;
+
+  @mustCallSuper
+  @override
+  Future<void> onLoad() async {
+    super.onLoad();
+    add(skinContainer);
+    add(labelAlignContainer);
+  }
+
+  @protected
+  final skinContainer = PositionComponent();
```

Comment:
Is this needed? Can't you just set the skins as children directly?

Reply:
Yes, because we want the skins to always be below the labels

Reply:
Alright, I guess this is easier than fiddling with all the priorities.
I think it should be possible to have it as a `Component` instead of a `PositionComponent` at least though?

Reply:
Yes, I can replace it with a Component.